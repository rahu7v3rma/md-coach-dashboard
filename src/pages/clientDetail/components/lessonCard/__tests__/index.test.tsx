import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';

import LessonCard from '..';

const mock = {
    imageUrl: 'imageUrl',
    title: 'title',
    subTitle: 'subTitle',
    time: 'time',
    onClick: jest.fn()
};

jest.mock('src/navigation');

test('Snapshot', () => {
    const view = render(<LessonCard {...mock} />);
    expect(view.asFragment()).toMatchSnapshot();
});

test('renders LessonCard component with all props', () => {
    const view = render(<LessonCard {...mock} />);
    expect(view).toBeTruthy();
});

test('renders LessonCard component with missing title, time and image props', () => {
    const view = render(<LessonCard lesson="" title="" />);
    expect(view.container.textContent).toBe('No Lesson Found');
});

test('handles image loading error', async () => {
    render(<LessonCard {...mock} />);
    let icon = screen.queryByAltText('icon');
    expect(icon).toBeInTheDocument();
    act(() => fireEvent.error(icon as HTMLElement));
    icon = screen.queryByAltText('icon');
    expect(icon).not.toBeInTheDocument();
});
