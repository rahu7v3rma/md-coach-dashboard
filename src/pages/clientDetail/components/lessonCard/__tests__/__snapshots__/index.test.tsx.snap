// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ispOId eJtfGw"
    height="100px"
    width="335px"
  >
    <img
      alt="icon"
      class="sc-cNNTdL enpNFh"
      data-testid="lesson_image"
      src="imageUrl"
    />
    <div
      class="sc-ehYhRg jxNZlb"
    >
      <p
        class="sc-jJTsDX iVQlaw"
        color="#271A51"
        data-testid="lesson_title"
        font-weight="700"
      >
        title
      </p>
      <p
        class="sc-jJTsDX eWOEaN"
        color="#A4AAAF"
        data-testid="lesson_subtitle"
        font-size="12"
        font-weight="300"
      >
        subTitle
      </p>
    </div>
    <div
      class="sc-fia-DRh czKmhu"
    >
      <p
        class="sc-jJTsDX iXagFl"
        color="#72996B"
        font-size="12"
      >
        time
      </p>
    </div>
  </div>
</DocumentFragment>
`;
